src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+public abstract class TwoPassVariantWalker extends VariantWalker {
```

Comment:
This is great - we can use this in the CNNScoreVariants tool (https://github.com/broadinstitute/gatk/issues/4316). It will need class and method javadoc though, and unit tests (see the ones for TwoPassReadWalker).